### PR TITLE
DPS-1578 - Migrate EFF library to HO-wide Artifactory and Drone.

### PR DESCRIPTION
**Pull request by noels**
- Add .drone.yml to build, test and publish to artifactory.
- Add .npmrc to enable publishting to any repo.
- Secrets are in Drone.
- Note the additional _ in the NPM_CONFIG__AUTH secret.
- Secrets are populated by drone and consumed by npm as environment variables.

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+always-auth = true
+email = "noel.sharpe@digital.homeoffice.gov.uk"
```
Comment (jaykeshur) [inline .npmrc]: Don't add this.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    commands:
+      - npm i
+      - npm test
+    when:
+      event: [push, pull_request, tag]
+
+  publish_to_artifactory:
+    image: node:6
+    secrets:
+      - npm_config__auth
```
Comment (jaykeshur) [inline .drone.yml:20]: two __

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+pipeline:
+
+  test:
+    image: node:6
+    secrets:
+      - npm_config__auth
```
Comment (jaykeshur) [inline .drone.yml:6]: two __

**Review by jaykeshur — CHANGES_REQUESTED** [review]

**Comment by noels** [comment]
Did you read the commit message?

File: .npmrc (hunk 1 above)
Comment (noels) [reply .npmrc]: email is required for npm to authenticate. I will put the value into a secret. What email address would  you suggest?

File: .drone.yml (hunk 2 above)
Comment (noels) [reply .drone.yml:20]: Two _ are required. The variable expected by npm is _auth not auth.  I included this in the commit message, which I'm sure you read ;)

File: .drone.yml (hunk 3 above)
Comment (noels) [reply .drone.yml:6]: Correct.

**Review by jaykeshur — APPROVED** [review]

**Comment by jaykeshur** [comment]
You probably want to change the drone event trigger to just push, rather than pull_request, as the secrets won't be exposed to the build then. 
Either that, or expose the secrets to the pull_request event (drone via the drone cli), but probably not necessary.

**Comment by davet1985** [comment]
This doesn't need to be moved to ACP artifactory, as it's an open source package with travis pipeline to NPM.

**Pull request closed without merging**